tests/asahirov/test_homework_7.py

```diff
@@ -0,0 +1,191 @@
+import random
+from operator import gt, lt, le, ge
+import time
+
+"""
+Create decorator which would remember answers,
+and if answer was given, do not make a call to a function
+"""
+
+
+def remember_answer(func):
+    answer = {}
+
+    def remember(*args):
+        if args in answer:
+            return answer[args]
```

Comment:
will fail if args contains mutable objects
though is good solution,

to provide more clear error you can check if args is hashable

```
try: 
    hash(args)
except Exception as e:
    raise TypeError("input args not hashable ")
```